packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -100,38 +105,141 @@ typedef WidgetTesterCallback = Future<void> Function(WidgetTester widgetTester);
 /// ```
 @isTest
 void testWidgets(
-  String description,
-  WidgetTesterCallback callback, {
-  bool skip = false,
-  test_package.Timeout timeout,
-  Duration initialTimeout,
-  bool semanticsEnabled = true,
-}) {
+    String description,
+    WidgetTesterCallback callback, {
+      bool skip = false,
+      test_package.Timeout timeout,
+      Duration initialTimeout,
+      bool semanticsEnabled = true,
+      Iterable<TestDimension<dynamic>> matrix = const <TestDimension<dynamic>>[],
+    }) {
   final TestWidgetsFlutterBinding binding = TestWidgetsFlutterBinding.ensureInitialized() as TestWidgetsFlutterBinding;
   final WidgetTester tester = WidgetTester._(binding);
-  test(
-    description,
-    () {
-      SemanticsHandle semanticsHandle;
-      if (semanticsEnabled == true) {
-        semanticsHandle = tester.ensureSemantics();
-      }
-      tester._recordNumberOfSemanticsHandles();
-      test_package.addTearDown(binding.postTest);
-      return binding.runTest(
-        () async {
-          debugResetSemanticsIdCounter();
-          await callback(tester);
-          semanticsHandle?.dispose();
-        },
-        tester._endOfTestVerifications,
-        description: description ?? '',
-        timeout: initialTimeout,
+  if (matrix == null || matrix.isEmpty) {
+    matrix = <DefaultDimension>[DefaultDimension()];
+  }
+  for (TestDimension<dynamic> dimension in matrix) {
+    for (dynamic variation in dimension.variations) {
```

Comment:
prefer `Object` to `dynamic`, avoids accidentally calling anything on `variation`